src/controller/CHIPDeviceController.cpp

```diff
@@ -250,11 +251,24 @@ void DeviceController::OnLocalNOCChainGeneration(void * context, CHIP_ERROR stat
         err = ConvertX509CertsToChipCertArray(noc, icac, chipCertSpan);
         SuccessOrExit(err);
 
-        err = fabric->SetOperationalCertsFromCertArray(chipCertSpan);
+        err = newFabric.SetOperationalCertsFromCertArray(chipCertSpan);
         SuccessOrExit(err);
     }
 
-    err = controller->mFabrics.Store(fabric->GetFabricIndex());
+    err = controller->mFabrics.AddNewFabric(newFabric, controller->mFabricIndex);
+    SuccessOrExit(err);
+    ChipLogProgress(Controller, "Added new fabric %d", controller->mFabricIndex);
+
+    ChipLogProgress(Controller, "Storing controller credentials");
+    err = controller->mFabrics.Store(controller->mFabricIndex);
+    SuccessOrExit(err);
+
+    fabric = controller->mFabrics.FindFabricWithIndex(controller->mFabricIndex);
```

Comment:
When would `fabric` be != to `newFabric` ?

Reply:
The `newFabric` object is on stack, and gets freed, including memory for certificates associated with it. The object retrieved from `mFabrics` is valid for the lifetime of the Fabric. Currently we are generating and storing the credentials using the fabric info. So using `newFabric` is not correct for that.

This will get reworked as we refactor commissioner's credentials out of this class.